src/plugins/credential_management/server/crypto/cli.ts

```diff
@@ -61,37 +69,30 @@ export class CryptoCli {
     return result.plaintext.toString();
   }
 
-  public static getInstance(): CryptoCli {
+  public static getInstance(path = defaultPath): CryptoCli {
     if (!CryptoCli._instance) {
-      CryptoCli._instance = new CryptoCli();
+      CryptoCli._instance = new CryptoCli(path);
     }
 
     return CryptoCli._instance;
   }
-
-  // TODO: Fine grain config loader
-  public static loadConfigAndGetPath(path: string): string {
-    const yaml = safeLoad(readFileSync(path, 'utf8'));
-    if (yaml !== null && typeof yaml === 'object') {
-      return JSON.parse(JSON.stringify(yaml))['multiDataSource.materialPath'];
-    }
-    // console.error('Load failed! Please check the config path.');
-    // Return Default Path
-    return './crypto_material';
-  }
-
-  // TODO: Support configurable crypto materials file path 
-  public static generateCryptoMaterials(keyName: string, keyNamespace: string) {
-    const cryptoMaterials = {
-      keyName,
-      keyNamespace,
-      unencryptedMasterKey: randomBytes(32),
-    };
-    const path = CryptoCli.loadConfigAndGetPath('config/credential_management.yml');
-
-    writeFile(path, JSON.stringify(cryptoMaterials), function (err) {
-      if (err) throw err;
-    });
-    // console.info('crypto materials generated successfully!');
-  }
 }
+
+const generateCryptoMaterials = function (
+  path = defaultPath,
+  keyName = 'keyName',
+  keyNamespace = 'keyNamespace'
```

Comment:
does this hard-coded keyName and keyNamespace impact the crypto materials?

Reply:
> does this hard-coded keyName and keyNamespace impact the crypto materials?

Yes, it's used to identify the keyring.

Reply:
so I'd suggest to allow them to be configured